### ISSUE TITLE
Barycentric Subdivision

### ISSUE THREAD

**Issue by prinshul** [aidos-lab/mantra#15]
How to compute Barycentric Subdivision as reported in the paper? Is there any demo or code?

Thanks.

**Comment by rballeba** [comment]
Dear @prinshul,

You can apply barycentric subdivisons using the transform `barycentric_subdivision_transform` located in the [`mantra-benchmarks` repo](https://github.com/aidos-lab/mantra-benchmarks/blob/5472fdef66d21f048528a616ab53078410a98b5d/code/datasets/simplicial.py#L21) on the PyTorch Data Representation of MANTRA. A small example of how to subdivide a simplicial complex using our code is the following one:

```
from mantra.datasets import ManifoldTriangulations
from datasets.simplicial import barycentric_subdivision_transform
num_subdivisions = 1
dataset = ManifoldTriangulations(root="./data", manifold="2", version="latest")
complex_to_subdivide = dataset[0]
BarycentricSubdivisionTransform = barycentric_subdivision_transform(num_subdivisions)
complex_subdivided = BarycentricSubdivisionTransform(complex_to_subdivide)
print(complex_to_subdivide)
print(complex_subdivided)
```

The output obtained is a `torch_geometric.data.data.Data` containing information about the simplicial complex such as Betti numbers, torsion coefficients, and a copy of the simplicial complex in [`TopoNetX`](https://github.com/pyt-team/TopoNetX) format. Notice that you do not obtain features on the subdivision, and to get features, you should use the transformations provided in the mantra-benchmark repository.

**Comment by prinshul** [comment]
Thank you

**Comment by prinshul** [comment]
Are the 2_M_H_1 numbers reported on models trained only on 2_M_H_0?

How are the results reported for barycentric subdivision, a model trained on 2_M_H_0 tested on 2_M_H_1 (with barycentric subdivision)?

**Comment by danielbinschmid** [comment]
Dear Prinshul,
- 2_M_H_1  are trained on 2_M_H_0.
- Trained on 2_M_H_0  and tested on 2_M_H_1.
- yes, 2_M_H_1  is a test set with a barycentric subd.

 2_M_H_0 => 2d manifold with homomorphism type known. >>>> Correct
 2_M_0 => 2d manifold with homomorphism type both unknown and known. >>>> Correct
 2_M_H_1 => test set of 2d manifold with homomorphism type known with barycentric subdivision. >>>> Correct
